src/components/TradeAssetSearch/hooks/useGetCustomTokensQuery.tsx

```diff
@@ -42,18 +48,50 @@ export const useGetCustomTokensQuery = ({
     [contractAddress],
   )
 
+  const getSolanaTokenMetadata = useCallback(
+    async (mintAddress: string): Promise<TokenMetadata> => {
+      const solanaRpcUrl = `${getConfig().REACT_APP_ALCHEMY_SOLANA_BASE_URL}/${
```

Comment:
Summoning @NeOMakinG here - think we already have a way to fetch token metadata from somewhere else?

Reply:
We are using coingecko, so if asset is not inside the asset file, we don't have any other way to fetch it properly with icon etc

We could fetch it using the token2022/SPL programs but we wouldn't have any icon, wdyt?

Reply:
Sounds sane to me then - icon is definitely required for things to look pretty!